posturemanagementv1/posture_management_v1_integration_test.go

```diff
@@ -0,0 +1,326 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package posturemanagementv1_test
+
+import (
+	"fmt"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/scc-go-sdk/posturemanagementv1"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"os"
+)
+
+/**
+ * This file contains an integration test for the posturemanagementv1 package.
+ *
+ * Notes:
+ *
+ * The integration test will automatically skip tests if the required config file is not available.
+ */
+
+var _ = Describe(`PostureManagementV1 Integration Tests`, func() {
+
+	const externalConfigFile = "../posture_management_v1.env"
+
+	var (
+		scopeId                  string
+		collectorId              string
+		credentialId             string
+		scanId                   string
+		credentialString         string
+		err                      error
+		accountId                string
+		apiKey                   string
+		authUrl                  string
+		authenticator            core.IamAuthenticator
+		postureManagementService *posturemanagementv1.PostureManagementV1
+		serviceURL               string
+		config                   map[string]string
+	)
+
+	var shouldSkipTest = func() {
+		Skip("External configuration is not available, skipping tests...")
+	}
+
+	Describe(`External configuration`, func() {
+		It("Successfully load the configuration", func() {
+			_, err = os.Stat(externalConfigFile)
+			if err != nil {
+				Skip("External configuration file not found, skipping tests: " + err.Error())
+			}
+
+			os.Setenv("IBM_CREDENTIALS_FILE", externalConfigFile)
+			config, err = core.GetServiceProperties(posturemanagementv1.DefaultServiceName)
+			if err != nil {
+				Skip("Error loading service properties, skipping tests: " + err.Error())
+			}
+
+			accountId = config["POSTURE_ACCOUNT_ID"]
+			if accountId == "" {
+				Skip("Unable to load posture account configuration property, skipping tests")
+			}
+
+			serviceURL = config["URL"]
+			if serviceURL == "" {
+				Skip("Unable to load service URL configuration property, skipping tests")
+			}
+
+			authUrl = config["IAM_APIKEY_URL"]
+			if authUrl == "" {
+				Skip("Unable to load auth service URL configuration property, skipping tests")
+			}
+
+			apiKey = config["IAM"]
+			if apiKey == "" {
+				Skip("Unable to load IAM configuration property, skipping tests")
+			}
+
+			credentialString = "{\"credential_name\":\"SDK CREDENTIALS PIPELINE TEST\",\"credential_description\":\"DEMO CREDENTIALS\",\"purpose\":\"discovery_fact_collection_remediation\",\"enabled\":true,\"credential_group\":{},\"credential_type\":\"ibm_cloud\",\"ibm_cloud\":{\"ibm_api_key\":\"lkjb1rsBsqGKyPBXL7YRwmd-JUl9fI38Ntym0RarZGD-\"}}"
```

Comment:
API key should be read from ENV or a file

Reply:
you can read it from a ENV var and set it up in CI ... i wonder why didn't you receive ServiceNow ticket for this 😅

Reply:
> API key should be read from ENV or a file

accepted and updated